strawberryfields/backends/tfbackend/ops.py

```diff
@@ -159,44 +149,27 @@ def squeezed_vacuum_vector(r, theta, D, batched=False, eps=1e-32):
     return output
 
 
-def squeezer_matrix(r, theta, D, batched=False):
-    """creates the single mode squeeze matrix"""
-    r = tf.cast(r, tf.float64)
-    if not batched:
-        r = tf.expand_dims(r, 0)  # introduce artificial batch dimension
-    r = tf.reshape(r, [-1, 1, 1, 1])
-    theta = tf.cast(theta, def_type)
-    theta = tf.reshape(theta, [-1, 1, 1, 1])
+@tf.custom_gradient
+def single_squeezing_matrix(r, phi, D, dtype=def_type.as_numpy_dtype):
```

Comment:
(Minor suggestion): could be nice to move `dtype=def_type.as_numpy_dtype` within the function:
```python
def single_squeezing_matrix(r, phi, D, dtype=None):
    if dtype is None:
        dtype=def_type.as_numpy_dtype
```
This would allow for example that the casting is not missed.

Reply:
good idea!